File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l=1;
        int r=numbers.size();
        while(l<r){
            int sum=numbers[l-1]+numbers[r-1];
            if(sum==target){
                return {l,r};
            }
            else if(sum>target)r--;
            else l++;
        }
        return {};
        
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        mp={}
        for i in range(len(numbers)):
            complement=target-numbers[i]
            if complement in mp:
                return [mp[complement]+1,i+1]
            mp[numbers[i]]=i    
        return []